parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -212,8 +212,20 @@ private static MetadataFilter filter(boolean skipRowGroups) {
   }
 
   private static <T> List<T> runAllInParallel(int parallelism, List<Callable<T>> toRun) throws ExecutionException {
-    LOG.info("Initiating action with parallelism: {}", parallelism);
-    ExecutorService threadPool = Executors.newFixedThreadPool(parallelism);
+    if (toRun.isEmpty()) {
+      return Collections.emptyList();
+    } else if (toRun.size() == 1) {
```

Comment:
Fine with me. 1 thread instead of 5 with the default config is still beneficial. Shall I update the PR with this change or you have more suggestions?